caddyhttp/httpserver/server.go

```diff
@@ -356,6 +356,16 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	c := context.WithValue(r.Context(), OriginalURLCtxKey, urlCopy)
 	r = r.WithContext(c)
 
+	// Setup a response recorder for this request.
+	rr := NewResponseRecorder(w)
+	w = rr
+
+	// Setup a replacer for the request that keeps track of placeholder
+	// values across plugins.
+	replacer := NewReplacer(r, rr, "")
```

Comment:
The `log` directive (for example) uses an empty value of `"-"`. How can we make this something customizable per-middleware?

Reply:
In the body of `NewReplacer`, I maintain some functions from the replacer for the request, but not all. The `emptyValue` is one of the things that gets set every time, so I'm pretty sure that this functionality it preserved. I'll see if a test can be crafted for that.